sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/NamedRelation.scala

```diff
@@ -21,4 +21,7 @@ import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 
 trait NamedRelation extends LogicalPlan {
   def name: String
+
+  // When true, the schema of input data must match the schema of this relation, during write.
+  def requireSchemaMatch: Boolean = true
```

Comment:
Separate from the discussion about the method name, I don't think it makes sense for this to be in `NamedRelation`.

`NamedRelation` exists to help create better error messages from the generic rules that apply across any relation. This addition doesn't fit with that purpose. I think the reason why this was added to `NamedRelation` is because the v2 relation class is not available to catalyst, but this rule is in catalyst. If that's the case, then this depends on moving v2 into catalyst and I think it makes sense to do that first.

Reply:
Moving DS v2 to catalyst module should be done soon, we can wait for it. BTW do we still need `NamedRelation` after that? It looks to me that `NamedRelation` is mostly for testing. Currently only the v2 relation class extends it.

Reply:
We probably don't need it any more.

Reply:
@cloud-fan, we may want to keep `NamedRelation` to be able to use `UnresolvedRelation` in v2 plans. If we updated `AppendData` (for example) to use `DataSourceV2Relation`, then we would not be able to create it with an `UnresolvedRelation` and delegate resolving the table to the analyzer.